### PR TITLE
Networking dio

### PR DESCRIPTION
**Pull request by dmitry-rachko-idf**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
+{
```
Comment (andrei-lastochkin-idf) [inline .vscode/launch.json:1]: I'm not sure if this file should not be in .gitignore. Check it please

File: android/app/build.gradle [hunk 2]
```diff
@@ -65,4 +66,5 @@ flutter {
 
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation platform('com.google.firebase:firebase-bom:29.1.0')
```
Comment (andrei-lastochkin-idf) [inline android/app/build.gradle:69]: u already can remove your firebase storage as we've got endpoints

Comment (andrei-lastochkin-idf) [inline android/app/build.gradle:69]: u already can remove your firebase storage as we've got endpoints

File: android/app/google-services.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
```
Comment (andrei-lastochkin-idf) [inline android/app/google-services.json:1]: -//-

Comment (andrei-lastochkin-idf) [inline android/app/google-services.json:1]: -//-

File: android/build.gradle [hunk 4]
```diff
@@ -8,6 +8,7 @@ buildscript {
     dependencies {
         classpath 'com.android.tools.build:gradle:4.1.0'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+        classpath 'com.google.gms:google-services:4.3.10'
```
Comment (andrei-lastochkin-idf) [inline android/build.gradle:11]: -//-

Comment (andrei-lastochkin-idf) [inline android/build.gradle:11]: -//-

File: packages/data/lib/di/data_injector.dart [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+void _initApiModule() {
+  GetIt.I.registerFactory<CancelToken>(() => CancelToken());
+  _initDioModule();
+  _initServiceModule();
+}
+
+void _initRepositoryModule() {
+  GetIt.I.registerFactory<INetworkRepository>(
```
Comment (andrei-lastochkin-idf) [inline packages/data/lib/di/data_injector.dart:21]: I would rather use singleton with repos

Comment (andrei-lastochkin-idf) [inline packages/data/lib/di/data_injector.dart:21]: I would rather use singleton with repos

File: packages/data/lib/dio/dio_builder.dart [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import 'package:dio/dio.dart';
+
+Dio dioBuilder(
+  String baseUrl, {
+  Map<String, dynamic>? headers,
+}) {
+  final timeout = 60000;
+  final headersMap = headers ?? <String, dynamic>{};
+  headersMap[HttpHeaders.userAgentHeader] =
+      '${Platform.operatingSystem}-APPLICATION'.toUpperCase();
```
Comment (andrei-lastochkin-idf) [inline packages/data/lib/dio/dio_builder.dart:12]: Are u sure u need this header now?

Comment (andrei-lastochkin-idf) [inline packages/data/lib/dio/dio_builder.dart:12]: Are u sure u need this header now?

File: packages/data/lib/repository/network_repository.dart [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+  @override
+  Future<RegistrationResponse> getRegistration() => _service
+      .get(
+        path: 'api/json?pretty=true',
+        cancelToken: _cancelToken,
+      )
+      .then(
+        (value) => Future.value(ApiRegistrationResponse.fromJson(value.data)),
+      );
```
Comment (andrei-lastochkin-idf) [inline packages/data/lib/repository/network_repository.dart:27]: I think this request is senseless without auth header

Comment (andrei-lastochkin-idf) [inline packages/data/lib/repository/network_repository.dart:27]: I think this request is senseless without auth header

File: packages/domain/lib/model/api_registration_response.dart [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import 'dart:convert';
+
+import 'registration_response.dart';
+
+class ApiRegistrationResponse implements RegistrationResponse {
+  @override
+  final Map<String, dynamic>? data;
+  @override
+  final String? step;
```
Comment (andrei-lastochkin-idf) [inline packages/domain/lib/model/api_registration_response.dart:9]: do u need this field?

Comment (andrei-lastochkin-idf) [inline packages/domain/lib/model/api_registration_response.dart:9]: do u need this field?

File: packages/domain/lib/model/registration_response.dart [hunk 9]
```diff
@@ -0,0 +1,4 @@
+abstract class RegistrationResponse {
+  Map<String, dynamic>? get data;
+  String? get step;
```
Comment (andrei-lastochkin-idf) [inline packages/domain/lib/model/registration_response.dart:3]: -//-

Comment (andrei-lastochkin-idf) [inline packages/domain/lib/model/registration_response.dart:3]: -//-

File: packages/domain/lib/repository/network_repository.dart [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import 'package:domain/model/registration_response.dart';
+
+import 'base_repository.dart';
+
+abstract class INetworkRepository implements BaseRepository {
+  Future<RegistrationResponse> getRegistration();
```
Comment (andrei-lastochkin-idf) [inline packages/domain/lib/repository/network_repository.dart:6]: -//-

Comment (andrei-lastochkin-idf) [inline packages/domain/lib/repository/network_repository.dart:6]: -//-

File: packages/domain/lib/usecase/get_registration_usecase.dart [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+
+  @override
+  Future<String?> call() async =>
+      _repository.getRegistration().then((value) => value.step);
+
+  @override
+  void dispose() {
+    _repository.dispose();
+  }
+}
```
Comment (andrei-lastochkin-idf) [inline packages/domain/lib/usecase/get_registration_usecase.dart:18]: -//-

Comment (andrei-lastochkin-idf) [inline packages/domain/lib/usecase/get_registration_usecase.dart:18]: -//-

File: packages/presentation/lib/screens/login_page/login_bloc.dart [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+
+  void sendRegistration();
+
+  void onEmailChanged(String email);
+
+  void onPasswordChanged(String password);
+}
+
+class _LoginBloc extends BlocImpl implements LoginBloc {
+  final GetRegistrationStepUseCase _getRegistrationStepUseCase;
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/screens/login_page/login_bloc.dart:28]: -//-

Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/screens/login_page/login_bloc.dart:28]: -//-

File: packages/presentation/lib/screens/login_page/login_bloc.dart [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+    _updateData();
+  }
+
+  @override
+  void sendRegistration() async {
+    _screenData.registrationStep = 'Loading';
+    _updateData();
+
+    final request = _viewMapper.mapScreenDataToRequest(_screenData);
+    await _sendRegistrationStepUseCase.call(request);
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/screens/login_page/login_bloc.dart:53]: u can simplify it with await _sendRegistrationStepUseCase(request). It's not a mistake, just note

File: packages/presentation/lib/screens/login_page/login_bloc.dart [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+  @override
+  void initState() {
+    super.initState();
+    _updateData();
+  }
+
+  @override
+  void sendRegistration() async {
+    _screenData.registrationStep = 'Loading';
+    _updateData();
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/screens/login_page/login_bloc.dart:50]: do u need this field if u have isLoading field in your BlocData?

File: packages/presentation/lib/screens/login_page/login_bloc.dart [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+
+  @override
+  void initState() {
+    super.initState();
+    _updateData();
+  }
+
+  @override
+  void sendRegistration() async {
+    _screenData.registrationStep = 'Loading';
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/screens/login_page/login_bloc.dart:49]: do u need this field if u have isLoading field in your BlocData?

Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/screens/login_page/login_bloc.dart:49]: do u need this field if u have isLoading field in your BlocData?

File: packages/presentation/lib/screens/login_page/login_page.dart [hunk 16]
```diff
@@ -0,0 +1,148 @@
+import 'package:flutter/material.dart';
+import 'package:presentation/bloc/bloc_data.dart';
+import 'package:presentation/screens/login_page/login_bloc.dart';
+import 'package:presentation/screens/login_page/login_data.dart';
+
+import '../../bloc/bloc_state.dart';
+// import '../root/root.dart';
+
+// import 'dart:developer' as developer;
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/screens/login_page/login_page.dart:9]: clean it

Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/screens/login_page/login_page.dart:9]: clean it

**Review by andrei-lastochkin-idf — CHANGES_REQUESTED** [review]

File: packages/domain/lib/model/api_registration_response.dart (hunk 8 above)
Comment (dmitry-rachko-idf) [reply packages/domain/lib/model/api_registration_response.dart]: nope

**Review by andrei-lastochkin-idf — APPROVED** [review]

**Pull request merged**